On creating challenge: "invalid memory address or nil pointer dereference"

**Describe the bug**:
`cert-manager`-pod crashes when issuing a new certificate with `dns01`-solver.

**Expected behaviour**:
Pod does not crash and certificate is created.

**Steps to reproduce the bug**:
Create an ingress-resource which points to an internal Load Balancer

**Anything else we need to know?**:
I have a private GKE-cluster with an internal nginx-ingress.
It does work with an external Load Balancer and a "normal" nginx-ingress.

Clusterissuerconfiguration:
```yaml
apiVersion: cert-manager.io/v1
kind: ClusterIssuer
metadata:
  name: letsencrypt-prod
  namespace: cert-manager
spec:
  acme:
    server: https://acme-v02.api.letsencrypt.org/directory
    email: example@example.com
    privateKeySecretRef:
      name: letsencrypt-prod
    solvers:
      - dns01:
          cloudDNS:
            project: the-project-id
            serviceAccountSecretRef:
              name: clouddns-dns01-solver-svc-acct
              key: key.json
```
Logs of crashed pod:
```logs
E1017 14:45:37.151855       1 util.go:71] cert-manager/controller/certificaterequests/handleOwnedResource "msg"="error getting referenced owning resource" "error"="certificaterequest.cert-manager.io \"unoconv-webservice-internal-prod-5lcww\" not found" "related_resource_kind"="CertificateRequest" "related_resource_name"="unoconv-webservice-internal-prod-5lcww" "related_resource_namespace"="default" "resource_kind"="Order" "resource_name"="unoconv-webservice-internal-prod-5lcww-3735918609" "resource_namespace"="default" "resource_version"="v1"
E1017 14:45:37.346965       1 controller.go:158] cert-manager/controller/orders "msg"="re-queuing item  due to error processing" "error"="ACME client for issuer not initialised/available" "key"="default/unoconv-webservice-internal-prod-5lcww-3735918609"
I1017 14:45:37.347217       1 setup.go:170] cert-manager/controller/clusterissuers "msg"="skipping re-verifying ACME account as cached registration details look sufficient" "related_resource_kind"="Secret" "related_resource_name"="letsencrypt-prod" "related_resource_namespace"="cert-manager" "resource_kind"="ClusterIssuer" "resource_name"="letsencrypt-prod" "resource_namespace"="" "resource_version"="v1"
E1017 14:45:37.832900       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 141 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1c0f800, 0x2df71b0)
        external/io_k8s_apimachinery/pkg/util/runtime/runtime.go:74 +0xa6
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        external/io_k8s_apimachinery/pkg/util/runtime/runtime.go:48 +0x89
panic(0x1c0f800, 0x2df71b0)
        GOROOT/src/runtime/panic.go:969 +0x175
github.com/jetstack/cert-manager/pkg/issuer/acme/dns/clouddns.(*DNSProvider).getHostedZone(0xc0005774a0, 0xc0001c6400, 0x36, 0xc000272680, 0xc0001c6380, 0x36, 0xc0001c6400)
        pkg/issuer/acme/dns/clouddns/clouddns.go:243 +0x2b7
github.com/jetstack/cert-manager/pkg/issuer/acme/dns/clouddns.(*DNSProvider).CleanUp(0xc0005774a0, 0xc001090db0, 0x25, 0xc0001c6400, 0x36, 0xc001090de0, 0x2b, 0x36, 0x0)
        pkg/issuer/acme/dns/clouddns/clouddns.go:191 +0x5a
github.com/jetstack/cert-manager/pkg/issuer/acme/dns.(*Solver).CleanUp(0xc0001423c0, 0x21a3800, 0xc000f43170, 0x21cdc80, 0xc000df4000, 0xc00109e600, 0xf, 0xc000501969)
        pkg/issuer/acme/dns/dns.go:160 +0x422
github.com/jetstack/cert-manager/pkg/controller/acmechallenges.(*controller).handleFinalizer(0xc0001fe000, 0x21a3800, 0xc000f43050, 0xc00109e600, 0x0, 0x0)
        pkg/controller/acmechallenges/sync.go:283 +0x328
github.com/jetstack/cert-manager/pkg/controller/acmechallenges.(*controller).Sync(0xc0001fe000, 0x21a3800, 0xc000f43050, 0xc00109e000, 0x0, 0x0)
        pkg/controller/acmechallenges/sync.go:68 +0x15a5
github.com/jetstack/cert-manager/pkg/controller/acmechallenges.(*controller).ProcessItem(0xc0001fe000, 0x21a3740, 0xc0003f0040, 0xc00060eb90, 0x43, 0x7fca2c110dd8, 0xc000f4c6c0)
        pkg/controller/acmechallenges/controller.go:204 +0x297
github.com/jetstack/cert-manager/pkg/controller.(*controller).worker.func1(0xc0003905a0, 0x1b6c540, 0xc0009ce2c0, 0xc0007cee08, 0x21b09e0, 0xc00041f740, 0x21a3740, 0xc0003f0040)
        pkg/controller/controller.go:157 +0x22f
github.com/jetstack/cert-manager/pkg/controller.(*controller).worker(0xc0003905a0, 0x21a3740, 0xc0003f0040)
        pkg/controller/controller.go:164 +0x105
github.com/jetstack/cert-manager/pkg/controller.(*controller).Run.func1()
        pkg/controller/controller.go:112 +0x67
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000f42db0)
        external/io_k8s_apimachinery/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000f42db0, 0x215b9c0, 0xc000f42ea0, 0x1, 0xc0001148a0)
        external/io_k8s_apimachinery/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000f42db0, 0x3b9aca00, 0x0, 0x1f7c601, 0xc0001148a0)
        external/io_k8s_apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(0xc000f42db0, 0x3b9aca00, 0xc0001148a0)
        external/io_k8s_apimachinery/pkg/util/wait/wait.go:90 +0x4d
created by github.com/jetstack/cert-manager/pkg/controller.(*controller).Run
        pkg/controller/controller.go:110 +0x1a9
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x16ac977]

goroutine 141 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        external/io_k8s_apimachinery/pkg/util/runtime/runtime.go:55 +0x10c
panic(0x1c0f800, 0x2df71b0)
        GOROOT/src/runtime/panic.go:969 +0x175
github.com/jetstack/cert-manager/pkg/issuer/acme/dns/clouddns.(*DNSProvider).getHostedZone(0xc0005774a0, 0xc0001c6400, 0x36, 0xc000272680, 0xc0001c6380, 0x36, 0xc0001c6400)
        pkg/issuer/acme/dns/clouddns/clouddns.go:243 +0x2b7
github.com/jetstack/cert-manager/pkg/issuer/acme/dns/clouddns.(*DNSProvider).CleanUp(0xc0005774a0, 0xc001090db0, 0x25, 0xc0001c6400, 0x36, 0xc001090de0, 0x2b, 0x36, 0x0)
        pkg/issuer/acme/dns/clouddns/clouddns.go:191 +0x5a
github.com/jetstack/cert-manager/pkg/issuer/acme/dns.(*Solver).CleanUp(0xc0001423c0, 0x21a3800, 0xc000f43170, 0x21cdc80, 0xc000df4000, 0xc00109e600, 0xf, 0xc000501969)
        pkg/issuer/acme/dns/dns.go:160 +0x422
github.com/jetstack/cert-manager/pkg/controller/acmechallenges.(*controller).handleFinalizer(0xc0001fe000, 0x21a3800, 0xc000f43050, 0xc00109e600, 0x0, 0x0)
        pkg/controller/acmechallenges/sync.go:283 +0x328
github.com/jetstack/cert-manager/pkg/controller/acmechallenges.(*controller).Sync(0xc0001fe000, 0x21a3800, 0xc000f43050, 0xc00109e000, 0x0, 0x0)
        pkg/controller/acmechallenges/sync.go:68 +0x15a5
github.com/jetstack/cert-manager/pkg/controller/acmechallenges.(*controller).ProcessItem(0xc0001fe000, 0x21a3740, 0xc0003f0040, 0xc00060eb90, 0x43, 0x7fca2c110dd8, 0xc000f4c6c0)
        pkg/controller/acmechallenges/controller.go:204 +0x297
github.com/jetstack/cert-manager/pkg/controller.(*controller).worker.func1(0xc0003905a0, 0x1b6c540, 0xc0009ce2c0, 0xc0007cee08, 0x21b09e0, 0xc00041f740, 0x21a3740, 0xc0003f0040)
        pkg/controller/controller.go:157 +0x22f
github.com/jetstack/cert-manager/pkg/controller.(*controller).worker(0xc0003905a0, 0x21a3740, 0xc0003f0040)
        pkg/controller/controller.go:164 +0x105
github.com/jetstack/cert-manager/pkg/controller.(*controller).Run.func1()
        pkg/controller/controller.go:112 +0x67
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000f42db0)
        external/io_k8s_apimachinery/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000f42db0, 0x215b9c0, 0xc000f42ea0, 0x1, 0xc0001148a0)
        external/io_k8s_apimachinery/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000f42db0, 0x3b9aca00, 0x0, 0x1f7c601, 0xc0001148a0)
        external/io_k8s_apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(0xc000f42db0, 0x3b9aca00, 0xc0001148a0)
        external/io_k8s_apimachinery/pkg/util/wait/wait.go:90 +0x4d
created by github.com/jetstack/cert-manager/pkg/controller.(*controller).Run
        pkg/controller/controller.go:110 +0x1a9
```

**Environment details:**:
- Kubernetes version: 1.16.13-gke.401
- Cloud-provider/provisioner: GKE
- cert-manager version: v1.0.3
- Install method: helm (cert-manager-v1.0.3)

/kind bug

--- a/pkg/issuer/acme/dns/clouddns/clouddns.go
+++ b/pkg/issuer/acme/dns/clouddns/clouddns.go
@@ -138,6 +138,7 @@ func NewDNSProviderServiceAccountBytes(project string, saBytes []byte, dns01Name
 		client:           svc,
 		dns01Nameservers: dns01Nameservers,
 		hostedZoneName:   hostedZoneName,
+		log:              logf.Log.WithName("clouddns"),
 	}, nil
 }
 

